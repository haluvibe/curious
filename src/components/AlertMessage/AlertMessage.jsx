/* eslint-disable react/no-multi-comp */

import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Dialog, { DialogTitle, DialogContent } from "material-ui/Dialog";
import Typography from "material-ui/Typography";

const styles = {};

class AlertMessage extends React.Component {
  handleClose = () => {
    this.props.onClose(this.props.selectedValue);
  };

  handleListItemClick = value => {
    this.props.onClose(value);
  };

  render () {
    const { classes, onClose, selectedValue, ...other } = this.props;

    return (
      <Dialog
        onClose={this.handleClose}
        aria-labelledby="simple-dialog-title"
        {...other}
      >
        <DialogTitle id="simple-dialog-title">Invalid credentials</DialogTitle>
        <DialogContent>
          <Typography>Error Error Error</Typography>
        </DialogContent>
      </Dialog>
    );
  }
}

AlertMessage.propTypes = {
  classes: PropTypes.object.isRequired,
  onClose: PropTypes.func,
  selectedValue: PropTypes.string
};

const AlertMessageWrapped = withStyles(styles)(AlertMessage);

export default class extends React.Component {
  state = {
    open: false
  };

  handleClickOpen = () => {
    this.setState({
      open: true
    });
  };

  handleClose = value => {
    this.setState({ open: false });
  };

  render () {
    return (
      <AlertMessageWrapped
        selectedValue={this.props.error}
        open={this.state.open}
        onClose={this.handleClose}
      />
    );
  }
}
