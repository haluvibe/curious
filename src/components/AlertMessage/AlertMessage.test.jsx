import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import AlertMessage from "./AlertMessage";

test("AlertMessage component should render as expected", () => {
  const component = shallow(<AlertMessage />),
    tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
