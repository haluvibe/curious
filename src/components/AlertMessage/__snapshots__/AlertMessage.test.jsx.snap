// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessage component should render as expected 1`] = `
<WithStyles(AlertMessage)
  onClose={[Function]}
  open={false}
/>
`;
